File: perl5/generate-elevenlabs.pl
#!/usr/bin/perl
# generate-elevenlabs.mjs — portiert nach perl5
# Quelle: javascript, Onboarding@main:scripts/generate-elevenlabs.mjs
# Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use File::Path qw(make_path);
use File::Spec;
use JSON;
use LWP::UserAgent;
use HTTP::Request::Common;
use URI::URL;

# Umgebungsvariablen abrufen
my $key = $ENV{ELEVENLABS_API_KEY};
my $voiceId = $ENV{ELEVENLABS_VOICE_ID} || "JBFqnCBsd6RMkjVDRZzb";

# Fehler werfen, wenn API-Key fehlt
die "ELEVENLABS_API_KEY fehlt." unless $key;

my $text = "Neun Projekte. Zwei Plattformen. Ein Ort, an dem Ideen verbunden und weiterentwickelt werden.";

# HTTP-Client initialisieren
my $ua = LWP::UserAgent->new;

# POST-Request vorbereiten
my $url = "https://api.elevenlabs.io/v1/text-to-speech/${voiceId}?output_format=mp3_44100_128";

my $req = POST $url,
    Content_Type => 'application/json',
    'xi-api-key' => $key,
    Content => encode_json({
        text => $text,
        model_id => "eleven_multilingual_v2",
        voice_settings => {
            stability => 0.58,
            similarity_boost => 0.72,
            style => 0.18,
            use_speaker_boost => \1  # true in JSON
        }
    });

# Request senden
my $response = $ua->request($req);

# Fehler werfen, wenn Request fehlschlägt
die "ElevenLabs fehlgeschlagen: " . $response->code unless $response->is_success;

# Zielverzeichnis erstellen
my $audio_dir = File::Spec->catdir(qw(.. public audio));
make_path($audio_dir) or die "Konnte Verzeichnis nicht erstellen: $!";

# Audio-Datei speichern
my $audio_file = File::Spec->catfile($audio_dir, "project-narration.mp3");
open my $fh, '>', $audio_file or die "Konnte Datei nicht öffnen: $!";
print $fh $response->content;
close $fh;

# JSON-Datei mit Metadaten erstellen
my $meta_data = {
    model => "eleven_multilingual_v2",
    voiceId => $voiceId,
    characters => length($text),
    text => $text,
    output => "public/audio/project-narration.mp3"
};

my $json_text = to_json($meta_data, { pretty => 1 });

my $meta_file = File::Spec->catfile(qw(.. media-production elevenlabs-result.json));
open my $meta_fh, '>', $meta_file or die "Konnte Metadaten-Datei nicht öffnen: $!";
print $meta_fh $json_text;
close $meta_fh;

print "ElevenLabs abgeschlossen: " . length($text) . " Zeichen.\n";

File: python/generate-elevenlabs.py
#!/usr/bin/env python3
# generate-elevenlabs.mjs — portiert nach python
# Quelle: javascript, Onboarding@main:scripts/generate-elevenlabs.mjs
# Erzeugt: 2026-08-09 durch ABSTRACTIONS_MANAGER.py

import os
import json
import asyncio
import aiohttp
import aiofiles
from pathlib import Path

async def main():
    key = os.environ.get('ELEVENLABS_API_KEY')
    voice_id = os.environ.get('ELEVENLABS_VOICE_ID', 'JBFqnCBsd6RMkjVDRZzb')
    
    if not key:
        raise Exception('ELEVENLABS_API_KEY fehlt.')
    
    text = 'Neun Projekte. Zwei Plattformen. Ein Ort, an dem Ideen verbunden und weiterentwickelt werden.'
    
    url = f'https://api.elevenlabs.io/v1/text-to-speech/{voice_id}?output_format=mp3_44100_128'
    headers = {
        'xi-api-key': key,
        'Content-Type': 'application/json'
    }
    payload = {
        'text': text,
        'model_id': 'eleven_multilingual_v2',
        'voice_settings': {
            'stability': 0.58,
            'similarity_boost': 0.72,
            'style': 0.18,
            'use_speaker_boost': True
        }
    }
    
    async with aiohttp.ClientSession() as session:
        async with session.post(url, headers=headers, json=payload) as response:
            if response.status != 200:
                raise Exception(f'ElevenLabs fehlgeschlagen: {response.status}')
            
            audio_data = await response.read()
            
            # Create directories
            audio_dir = Path(__file__).parent.parent / 'public' / 'audio'
            audio_dir.mkdir(parents=True, exist_ok=True)
            
            # Write audio file
            audio_path = audio_dir / 'project-narration.mp3'
            async with aiofiles.open(audio_path, 'wb') as f:
                await f.write(audio_data)
            
            # Write JSON result
            result_data = {
                'model': 'eleven_multilingual_v2',
                'voiceId': voice_id,
                'characters': len(text),
                'text': text,
                'output': 'public/audio/project-narration.mp3'
            }
            
            media_production_dir = Path(__file__).parent.parent / 'media-production'
            media_production_dir.mkdir(parents=True, exist_ok=True)
            
            result_path = media_production_dir / 'elevenlabs-result.json'
            async with aiofiles.open(result_path, 'w') as f:
                await f.write(json.dumps(result_data, indent=2))
    
    print(f'ElevenLabs abgeschlossen: {len(text)} Zeichen.')

if __name__ == '__main__':
    asyncio.run(main())
